packages/sdk/src/serialization.ts

```diff
@@ -65,19 +65,27 @@ function serializeAccountTransactionHeader(
 /**
  * Serializes a map of account transaction signatures. If no signatures are provided,
  * then an error is thrown.
+ *
+ * If `serializeSignatureLength` is `true` (default), the lengths of the inner signatures
+ * are serialized. Otherwise, the length is not serialized.
  */
 export function serializeAccountTransactionSignature(
-    signatures: AccountTransactionSignature
+    signatures: AccountTransactionSignature,
+    serializeSignatureLength = true
 ): Buffer {
     if (Object.keys(signatures).length === 0) {
         throw new Error('No signatures were provided');
     }
 
     const putSignature = (signature: string) => {
         const signatureBytes = Buffer.from(signature, 'hex');
-        const length = Buffer.alloc(2);
-        length.writeUInt16BE(signatureBytes.length, 0);
-        return Buffer.concat([length, signatureBytes]);
+        if (serializeSignatureLength) {
+            const length = Buffer.alloc(2);
+            length.writeUInt16BE(signatureBytes.length, 0);
+            return Buffer.concat([length, signatureBytes]);
+        } else {
+            return Buffer.concat([Buffer.from([0]), signatureBytes]);
```

Comment:
**Request change:** This serialization is very specific to CIS-3, so I don't think it should be part of this function, probably better to have it as a separate function.
Also, the `0` prefix needs to be explained in a comment.

Reply:
Just to be sure I understand: you are suggesting copying the whole function and making a `serializeCIS3AccountTransactionSignature` that prepends the zero byte?

Reply:
yes, I think it would be better to keep the two serializations separate.